src/app-bundles/maps-bundle.js

```diff
@@ -9,79 +9,103 @@ import ScaleBar from "ol/control/ScaleLine";
 import BasemapPicker from "../ol-controls/basemap-picker";
 
 const actions = {
-  MAPS_INITIALIZED: `MAPS_INITIALIZED`,
-  MAPS_SHUTDOWN: `MAPS_SHUTDOWN`,
+  MAPS_INITIALIZED: "MAPS_INITIALIZED",
+  MAPS_SHUTDOWN: "MAPS_SHUTDOWN",
+  MAPS_LOADED: "MAPS_LOADED",
 };
 
-export default {
-  name: "maps",
+export default (() => {
+  const internalState = {
+    map: null,
+  };
 
-  getReducer: () => {
-    const initialData = {};
+  const result = {
+    name: "maps",
+  
+    getReducer: () => {
+      const initialData = {
+        _initializedMap: null,
+        _mapLoaded: false,
+      };
+  
+      return (state = initialData, { type, payload }) => {
+        switch (type) {
+          case actions.MAPS_INITIALIZED:
+          case actions.MAPS_SHUTDOWN:
+          case actions.MAPS_LOADED:
+            return Object.assign({}, state, payload);
+          default:
+            return state;
+        }
+      };
+    },
+  
+    doMapsInitialize: (key, el, options) => ({ dispatch, store }) => {
```

Comment:
For example, these action creators take a key argument. I think that's how we would get and set the right map from the internal state. If that makes sense, we can discuss at scrum.